### PR TITLE
Implement "Animals" plugin.

### PR DESCRIPTION
**Pull request by SerperiorBae**
- Realised that jsbeautifier adds spaces instead of hard tabs.
- Fix the help command
- Remove Zoo.

### CODE REVIEW

File: chat-plugins/animals.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                    
+                }
+                
+            });
+            
+        });
+        req.end();
+        
+    },
+    animalshelp: ['|raw|<div class="infobox"><strong>Animals Plugin by <font color="' + Gold.hashColor('darknightskies') + '">DarkNightSkies</font> & <font color="' + Gold.hashColor('kyvn') + '">Kyv.n(♥)</font><strong><br>' +
```
Comment (panpawn) [inline chat-plugins/animals.js]: this kind of help command doesn't support HTML

**Comment by panpawn** [comment]
still spaces instead of tabs :x

**Comment by panpawn** [comment]
still spaces 🚕

File: chat-plugins/animals.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+const http = require('http');
+
+exports.commands = {
+    animal: 'animals',
+    animals: function(target, room, user) {
+        let tarId = toId(target)
```
Comment (panpawn) [inline chat-plugins/animals.js]: missing semi colon
what if target is undefined

File: chat-plugins/animals.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            'otter': 'otter',
+            'dog': 'dog',
+            'bunny': 'bunny',
+            'pokemon': 'pokemon',
+            'kitten': 'kitten',
+            'puppy': 'puppy'
+            
+        };
+        let validTarget = validTargets[tarId];
+        if (room.id === 'lobby' && validTarget) return this.errorReply("This command cannot be broadcasted in the Lobby.");
```
Comment (panpawn) [inline chat-plugins/animals.js]: ``this.broadcasting`` checks if it's being broadcasted

File: chat-plugins/animals.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+                        self.errorReply("ERROR: Command has crashed [No Images found]");
+                        return room.update();
+                        
+                    } else {
+                        self.sendReplyBox(output);
+                        return room.update();
+                        }
+                    
+                } catch (e) {
+                    self.errorReply("ERROR: Command has crashed [Memory Overloaded!]");
```
Comment (panpawn) [inline chat-plugins/animals.js]: does it actually happen?

File: chat-plugins/animals.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
+'use strict';
```
Comment (panpawn) [inline chat-plugins/animals.js]: you can do a comment up top to give yourself credit if you want

File: chat-plugins/animals.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+            'cat': 'cat',
+            'otter': 'otter',
+            'dog': 'dog',
+            'bunny': 'bunny',
+            'pokemon': 'pokemon',
+            'kitten': 'kitten',
+            'puppy': 'puppy'
+            
+        };
+        let validTarget = validTargets[tarId];
```
Comment (panpawn) [inline chat-plugins/animals.js]: there should be a check if they do a target that's not one of the ``validTargets`` or something

File: chat-plugins/animals.js [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+                    let output = '<center><img src="' + data.data["image_url"] + '" width="400"></center>';
+                    if (!self.runBroadcast()) return;
+                    if (data.data["image_url"] === undefined) {
+                        self.errorReply("ERROR: Command has crashed [No Images found]");
+                        return room.update();
+                        
+                    } else {
+                        self.sendReplyBox(output);
+                        return room.update();
+                        }
```
Comment (panpawn) [inline chat-plugins/animals.js]: indentation is wrong

File: chat-plugins/animals.js (hunk 6 above)
Comment (SerperiorBae) [inline chat-plugins/animals.js]: This is provided below panpawn: 

See line 22

File: chat-plugins/animals.js (hunk 4 above)
Comment (SerperiorBae) [inline chat-plugins/animals.js]: Should rename; it's moreso that the command gets used too much on giphys end.

File: chat-plugins/animals.js (hunk 7 above)
Comment (SerperiorBae) [inline chat-plugins/animals.js]: I honestly don't see it; I might be blind but I was using the indent that git gave me lma0

File: chat-plugins/animals.js (hunk 6 above)
Comment (panpawn) [inline chat-plugins/animals.js]: setting something to ``undefined`` isn't a good idea generally speaking

**Comment by panpawn** [comment]
doing just ``/animal`` makes the help command show up twice

**Pull request merged**